edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Uploader/FileWriter.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Writes to a local file instead of uploading.
+    /// </summary>
+    public class FileWriter : IMetricsUpload
```

Comment:
mark it as sealed; and please rename it with a specific name.

Reply:
IMetricsUpload; this name is weird when implemented by FileWriter.  I suggest to rename as "IMetricsPublisher" and its method should be "PublishAsync".  what do you think?

Reply:
That makes sense. There might be other ways we publish these metrics in the future